cli_tools/common/utils/daisy/resource_labeler_test.go

```diff
@@ -139,11 +164,51 @@ func TestUpdateWorkflowImagesLabelled(t *testing.T) {
 		buildID, &existingLabels)
 	validateLabels(t, &(*w.Steps["cimg"].CreateImages).Images[1].Image.Labels, "gce-image-import",
 		buildID)
-
 	validateLabels(t, &(*w.Steps["cimg"].CreateImages).Images[2].Image.Labels,
 		"gce-image-import-tmp", buildID, &existingLabels)
 	validateLabels(t, &(*w.Steps["cimg"].CreateImages).Images[3].Image.Labels,
 		"gce-image-import-tmp", buildID)
+
+	validateLabels(t, &(*w.Steps["cimg"].CreateImages).ImagesBeta[0].Image.Labels, "gce-image-import",
+		buildID, &existingLabels)
+	validateLabels(t, &(*w.Steps["cimg"].CreateImages).ImagesBeta[1].Image.Labels, "gce-image-import",
+		buildID)
+	validateLabels(t, &(*w.Steps["cimg"].CreateImages).ImagesBeta[2].Image.Labels,
+		"gce-image-import-tmp", buildID, &existingLabels)
+	validateLabels(t, &(*w.Steps["cimg"].CreateImages).ImagesBeta[3].Image.Labels,
+		"gce-image-import-tmp", buildID)
+
+}
+
+func TestUpdateWorkflowImageStorageLocationSet(t *testing.T) {
+	buildID := "abc"
+
+	w := daisy.New()
+	existingLabels := map[string]string{"labelKey": "labelValue"}
+	w.Steps = map[string]*daisy.Step{
+		"cimg": {
+			CreateImages: &daisy.CreateImages{
+				ImagesBeta: []*daisy.ImageBeta{
+					{
+						Image: computeBeta.Image{
+							Name:   "final-image-1",
+							Labels: map[string]string{"labelKey": "labelValue"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	rl := createTestResourceLabeler(buildID, userLabels)
+	rl.ImageLocation = "europe-west5"
+
+	rl.LabelResources(w)
+
+	validateLabels(t, &(*w.Steps["cimg"].CreateImages).ImagesBeta[0].Image.Labels, "gce-image-import",
+		buildID, &existingLabels)
+
+	assert.Equal(t, "europe-west5", (*w.Steps["cimg"].CreateImages).ImagesBeta[0].Image.StorageLocations[0])
```

Comment:
Should the test fail if there are other labels in the list? Does the test not care?

Reply:
That's already checked using existingLabels

Reply:
The 'ImagesBeta[0].Image.StorageLocations' list, I mean. For example, to ensure that a duplicate doesn't sneak in there or something like that. A quick check would be to assert that the list has length one, I'd think.

Reply:
Added length==1 assertion